File: java/class-organizer/src/test/java/com/rbleggi/classorganizer/ScheduleRemoveTest.java
package com.rbleggi.classorganizer;

import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;

import static org.junit.jupiter.api.Assertions.*;

class ScheduleRemoveTest {

    @Test
    void testRemoveExistingSession() {
        var teacher = new Teacher("Alice");
        var schedule = new Schedule();
        var session = new ClassSession("1", "Math", LocalDateTime.of(2025, 10, 4, 9, 0), LocalDateTime.of(2025, 10, 4, 10, 0), teacher);

        schedule.addClassSession(session);
        assertTrue(schedule.removeClassSession(session));
        assertEquals(0, schedule.getClassSessions().size());
    }

    @Test
    void testRemoveNonExistingSession() {
        var teacher = new Teacher("Bob");
        var schedule = new Schedule();
        var session = new ClassSession("1", "Physics", LocalDateTime.of(2025, 10, 4, 9, 0), LocalDateTime.of(2025, 10, 4, 10, 0), teacher);

        assertFalse(schedule.removeClassSession(session));
    }

    @Test
    void testRemoveOneOfMultiple() {
        var teacher = new Teacher("Charlie");
        var schedule = new Schedule();
        var session1 = new ClassSession("1", "Math", LocalDateTime.of(2025, 10, 4, 9, 0), LocalDateTime.of(2025, 10, 4, 10, 0), teacher);
        var session2 = new ClassSession("2", "Chemistry", LocalDateTime.of(2025, 10, 4, 11, 0), LocalDateTime.of(2025, 10, 4, 12, 0), teacher);

        schedule.addClassSession(session1);
        schedule.addClassSession(session2);
        schedule.removeClassSession(session1);
        assertEquals(1, schedule.getClassSessions().size());
        assertTrue(schedule.getClassSessions().contains(session2));
    }
}

File: kotlin/class-organizer/src/test/kotlin/com/rbleggi/calssorganizer/ScheduleRemoveTest.kt
package com.rbleggi.calssorganizer

import java.time.LocalDateTime
import kotlin.test.Test
import kotlin.test.assertTrue
import kotlin.test.assertFalse
import kotlin.test.assertEquals

class ScheduleRemoveTest {
    @Test
    fun `removeClassSession returns true for existing`() {
        val schedule = Schedule()
        val teacher = Teacher("Alice")
        val session = ClassSession("1", "Math", LocalDateTime.of(2025, 1, 1, 9, 0), LocalDateTime.of(2025, 1, 1, 10, 0), teacher)
        schedule.addClassSession(session)
        assertTrue(schedule.removeClassSession(session))
    }

    @Test
    fun `removeClassSession returns false for missing`() {
        val schedule = Schedule()
        val teacher = Teacher("Alice")
        val session = ClassSession("1", "Math", LocalDateTime.of(2025, 1, 1, 9, 0), LocalDateTime.of(2025, 1, 1, 10, 0), teacher)
        assertFalse(schedule.removeClassSession(session))
    }
}
